packages/runtime/container-runtime/src/throttler.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Used to give increasing delay times for throttling a single functionality.
+ * Delay is based on previous attempts within specified time window, ignoring actual delay time.
+ */
+ export class Throttler {
```

Comment:
would be good to have unit tests for this, as the behaviors are non obvious, and could easily be broken

Reply:
I opened a separate issue #6472 for Throttler related comments, since this PR just moved them to a separate file.